asf_public_discourse_home_decarbonisation/analysis/FAQ_analysis/questions_eda_analysis.py

```diff
@@ -0,0 +1,155 @@
+"""
+from asf_public_discourse_home_decarbonisation.config.plotting_configs import (
+    set_plotting_styles,
+    NESTA_COLOURS,
+)
+from asf_public_discourse_home_decarbonisation.utils.plotting_utils import (
+    finding_path_to_font,
+)
+import os
+import pandas as pd
+import matplotlib.pyplot as plt
+import textwrap
+from matplotlib.ticker import MaxNLocator
+import argparse
+from asf_public_discourse_home_decarbonisation.config.plotting_configs import (
+    NESTA_COLOURS,
+)
+from asf_public_discourse_home_decarbonisation import PROJECT_DIR
+
+set_plotting_styles()
+font_path_ttf = finding_path_to_font("Averta-Regular")
+
+
+def create_argparser() -> argparse.ArgumentParser:
+    """
+    Creates an argument parser that can receive the following arguments:
+    - category: category or sub-forum (defaults to "119_air_source_heat_pumps_ashp")
+    - forum: forum (i.e. mse or bh) (defaults to "bh")
+    Returns:
+        argparse.ArgumentParser: argument parser
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--category",
+        help="Category or sub-forum",
+        default="119_air_source_heat_pumps_ashp",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--forum",
+        help="forum (i.e. mse or bh)",
+        default="bh",
+        type=str,
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def load_data(file_path: str) -> pd.DataFrame:
+    """
+    Loads data from the extracted questions CSV file into a pandas DataFrame.
+
+    Args:
+        file_path (str): The path to the CSV file.
+
+    Returns:
+        pd.DataFrame: The loaded data.
+    """
+    return pd.read_csv(file_path)
+
+
+def get_question_counts(df: pd.DataFrame, column_name: str = "Question") -> pd.Series:
+    """
+    Calculates the frequency of each unique question in a specified column.
+
+    Args:
+        df (pd.DataFrame): The DataFrame containing the questions.
+        column_name (str): The name of the column from which to count questions.
+
+    Returns:
+        pd.Series: A Series containing question counts, indexed by the question text.
+    """
+    return df[column_name].value_counts()
+
+
+def wrap_text_with_ellipsis(text: str, line_width: int, max_lines: int) -> str:
+    """
+    Wraps text to a specified line width and maximum number of lines, adding an ellipsis if truncated.
+
+    Args:
+        text (str): The text to wrap.
+        line_width (int): The maximum width of each line.
+        max_lines (int): The maximum number of lines.
+
+    Returns:
+        str: The wrapped (and possibly truncated) text with an ellipsis if necessary.
+    """
+    wrapped_text = textwrap.wrap(text, width=line_width)
+    if len(wrapped_text) > max_lines:
+        return "\n".join(wrapped_text[:max_lines]) + "..."
+    else:
+        return "\n".join(wrapped_text)
+
+
+def plot_question_counts(
+    question_counts: pd.Series, top_n: int = 5, figure_path: str = None
```

Comment:
instead of a top_n I would have any number of questions, as long as their frequency is above a certain value, let's say 10.